File: hroom-server/src/main/java/com/dnbias/hroom/service/ReservationService.java
package com.dnbias.hroom.service;


import java.util.List;

import com.dnbias.hroom.reservation.Reservation;

public interface ReservationService {
    Reservation saveReservation(Reservation reservation);

    List<Reservation> fetchReservationList();

    Reservation updateReservation(Reservation reservation, Long reservationId);

    void deleteReservationById(Long reservationId);
}

File: hroom-web/src/main/web/src/app/service/reservation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Reservation} from "../shared/models/reservation";

@Injectable({
  providedIn: 'root'
})
export class ReservationService {
  private apiEndPoint: string = 'http://localhost:8888/api/v1/reservation';

  constructor(private http: HttpClient) { }

  saveReservation(reservation: Reservation):any {
    return this.http.post(this.apiEndPoint, reservation);
  }

  fetchReservationList(){
    return this.http.get(this.apiEndPoint);
  }

  findReservationById(id: number){
    return this.http.get(`${this.apiEndPoint}/${id}`);
  }

  updateReservation(reservation: Reservation, id: number){
    return this.http.put(`${this.apiEndPoint}/${id}`, reservation);
  }

  deleteReservationById(id: number){
    return this.http.delete(`${this.apiEndPoint}/${id}`, { responseType: 'text' });
  }
}
